packages/react/src/components/table/Table.tsx

```diff
@@ -0,0 +1,345 @@
+    if (!renderIndexCol) {
+      return column.key !== indexKey;
+    }
+    return true;
+  }).key;
+
+  return (
+    <>
+      {(checkboxSelection || heading || (customActionButtons && customActionButtons.length > 0)) && (
+        <div className={styles.actionContainer}>
+          {heading && (
+            <div id={headingId} role="heading" aria-level={headingAriaLevel} className={styles.heading}>
+              {heading}
+            </div>
+          )}
+          {(checkboxSelection || (customActionButtons && customActionButtons.length > 0)) && (
+            <div className={styles.actionButtonContainer}>
+              {checkboxSelection && (
+                <>
+                  <Button
+                    onClick={() => {
+                      selectAllRows();
+                    }}
+                    variant="secondary"
+                    size="small"
+                    disabled={selectedRows.length === rows.length}
+                    className={styles.actionButton}
+                    data-testid={
+                      dataTestId ? `hds-table-select-all-button-${dataTestId}` : 'hds-table-select-all-button'
+                    }
+                  >
+                    {selectAllRowsText || 'Valitse kaikki rivit'}
+                  </Button>
+                  <Button
+                    onClick={() => {
+                      deSelectAllRows();
+                    }}
+                    variant="secondary"
+                    size="small"
+                    disabled={selectedRows.length === 0}
+                    className={styles.actionButton}
+                    data-testid={
+                      dataTestId ? `hds-table-deselect-all-button-${dataTestId}` : 'hds-table-deselect-all-button'
+                    }
+                  >
+                    {clearSelectionsText || 'Tyhjennä valinnat'}
+                  </Button>
+                </>
+              )}
+              {customActionButtons &&
+                customActionButtons.length > 0 &&
+                customActionButtons.map((actionButton) => {
+                  return actionButton;
+                })}
+            </div>
+          )}
+        </div>
+      )}
+      <TableContainer
+        variant={variant}
+        dataTestId={dataTestId}
+        dense={dense}
+        zebra={zebra}
+        verticalLines={verticalLines}
+        customThemeClass={customThemeClass}
+        headingId={heading ? headingId : undefined}
+        {...rest}
+      >
+        {caption && <caption className={styles.caption}>{caption}</caption>}
+        {verticalHeaders && verticalHeaders.length && <VerticalHeaderColGroup />}
+        <thead>
+          <HeaderRow>
+            {verticalHeaders && verticalHeaders.length && <td role="presentation" />}
+            {checkboxSelection && <td className={styles.checkboxHeader} />}
+            {cols.map((column) => {
+              if (column.key === indexKey && !renderIndexCol) {
```

Comment:
Would this condition be more apparent as a function or constant?
```const isHiddenIndexColumn = (column.key === indexKey && !renderIndexCol)```
Or would it be possible to remove it altogether from the columns?
```const visibleColumns =  renderIndexCol ? cols : cols.filter(column => (column.key === indexKeu))```

Reply:
Good idea:

- [ ] use visibleColumns to increase readability and decrease repetition